### PR TITLE
C++: Add slint::Window::scale_factor() getter

### LINKED ISSUE #3003: get current DPI of a window
Right now there is no API to get the current DPI used by a `slint::Window`.

### PR DESCRIPTION
**Pull request by tronical**
Fixes #3003

### CODE REVIEW

**Comment by tronical** [comment]
I’ll add a test that verifies the value is greater than zero.

**Review by ogoffart — APPROVED** [review]

File: api/cpp/tests/window.cpp [hunk 1]
```diff
@@ -28,3 +28,19 @@ TEST_CASE("Basic Window Visibility")
     instance->hide();
     REQUIRE(instance->window().is_visible() == false);
 }
+
+TEST_CASE("Window Scale Factory Existence")
+{
+    using namespace slint::interpreter;
```
Comment (ogoffart) [inline api/cpp/tests/window.cpp:34]: The window.cpp test is build unconditionally regardless of SLINT_FEATURE_INTERPRETER
So this should probably be fixed in the tests/CMakeLists.txt  (But that's not a new problem)

**Review by ogoffart — APPROVED** [review]

File: api/cpp/tests/window.cpp (hunk 1 above)
Comment (tronical) [reply api/cpp/tests/window.cpp:34]: Good point, I'll push a change to master for that.
